wiki/Publishing-guidelines/Writing-functional-specs.md

```diff
@@ -0,0 +1,383 @@
+
+```html
+- [[Component or wiki page name 1]]
+- [[Component or wiki page name 2]]
+- [[Component or wiki page name 3]]
+```
+
+> ![image](https://user-images.githubusercontent.com/3793636/117873919-f6faba80-b265-11eb-81a5-039bdcd822e8.png)  Notice the double brackets with the file name `[[File name]]` in-between. This is different from a markdown link `[text](url)`.
+
+<br />
+</details><br /><br />
+
+#### Used by
+
+By properly linking dependencies we will also run an automation script to collect all the dependencies and create a used by list in addition. Used by components once populated communicates to the reader where this component is used.
+
+###### How to add a used by
+
+> ![image](https://user-images.githubusercontent.com/3793636/117873919-f6faba80-b265-11eb-81a5-039bdcd822e8.png)  Keep dependencies up to date, and leverage used by template section in the component wiki page.
+
+#### Backlinks
+Every time a wiki document references a component other than itself it creates a back link. Those back links are listed out on that component's page as a snippet and categorized by pages. When clicked it will go to the section of the page where it was talked about, and highlight its contents. A backlink doesn't have to be used as a depency to work, and this list is automated. As long as you include it on the page it'll render its content if available.
+
+#### Resources
+The resource section is a numbered list of additional resources not included in the dependency list. For example design specs, box folders, prototypes, and maybe additional documentation are good things you could include here. Follow the numbering standard if possible to star consistent. `r` for resources, and `#` in the list. If for whatever reason we need to remove an item it's best to keep the numbers the name, and skip since there is possible cross page linking and to avoid multiple updates when a resource becomes outdated.
+
+<details>
+  <summary><strong>Example</strong></summary><br />
+
+```html
+- r1: [resource example](www.example.com)
+- r2: [resource example](www.example.com)
+- r3: [resource example](www.example.com)
+- r6: [resource example](www.example.com)
+- r7: [resource example](www.example.com)
+```
+
+<br />
+</details><br /><br />
+
+### Sections
+
+The sections on the page are just some ways I've used in the past to break content down, and provide the functional specs required for development. Ultimately the following levels use at your discretion, and discuss in review, but the following is how I tend to like to use them.
+
+<br /><br />
+
+#### Global
+The global level is often where I'll put a visual of the component, and a high level break down of its elements and components. Every once in while I'll add a global option table or a logic conditional table to demonstrate how whole entire component's functionality can be modified. I typically won't include a markdown title to help indicate that it's apart of the highest level of the component.
```

Comment:
```suggestion
Use Global to include a visual of the component, and a high level break down of its elements and components. Depending on the needs of the component, it may be helpful to include a global option table or a logic conditional table to demonstrate how the whole component's functionality can be modified. 
```

Reply:
@photodow I didn't understand the last sentence here. 

> I typically won't include a markdown title to help indicate that it's apart of the highest level of the component.

Reply:
Just trying to communicate that if this section is included there isn't typically a `### global` heading.

It's usually something like this at a high level...

```
# component name

{{meta}
{{global functions}}

## Level 1

{{level 1 functions}}
```